wallet/dapps-communication.md

```diff
@@ -8,4 +8,67 @@ Initial meta issue: <https://github.com/massalabs/massa-standards/issues/13>
 
 ## Specification
 
+Here is pseudo code for the two points of view:
+
+- browser extension developer
+- dapp builder
+
+**As a browser extension developer, I would write...**
+
+browser extension code:
+
+```text
+// the extension register itself
+window.massaWalletProvider.addEventListener('loaded', () => {
+  window.massaWalletProvider.emit('register', {
+    ProviderName: "bearby",
+    htmlObject: "window.bearyWalletProvider"
+  })
+})
+```
+
+**Code snippet of massa javascript (or typescript) library:**
+
+```text
+registeredProviders = []
+
+// we start by listening register event
+window.massaWalletProvider.addEventListener('register', (payload) => {
+  registeredProviders.push(payload)
+})
+
+window.massaWalletProvider.emit('loaded') // all wallet will catch this event and emit register event
+```
+
+**As a dapp builder, I would write...**
+
+```text
+providers Provider[] = await massa-library.listWalletProviders()
+
+for each providers as p
+  for each p.listWallets as wallet
+    print wallet.getAddress()
+    print wallet.getBalance()
+
+wallet = providers.listWallets()[0]
+pkey, signature = await wallet.sign([...])
+```
+
+**Code snippet of massa javascript (or typescript) library:**
+
+```text
+export function listWalletProviders() {...}
+
+export class Wallet {
+  getAddress() {...}
+  getBalance() {...}
+  sign(payload: bytes): (string, bytes) {...}
+}
+
+export class Provider {
```

Comment:
One need is mising from https://github.com/massalabs/massa-standards/issues/13 :

```
a way to manage the wallet if authorized by the user (add/remove keys)
```

Reply:
Here I would refer to what Greg is saying in terms of missing funcs: https://github.com/massalabs/massa-standards/pull/26/files#diff-dad455c0814fa2229eaae244a61b26f2f312670c0d14b6da452cd46744739956R137-R175
Above are some more examples of what we could/must have a wallet expose.

Reply:
No I don't think that all the functions listed there are needed now. Can you extract the meaningful ones and complete this pseudo code section ?

Reply:
```typescript
export class Provider {
  // shows the connected network
  getConnectedNetwork(): MassaNetwork;
  // connects to the provider
  async connect(massaNetwork: MassaNetwork);
  // disconnects from the provider
  async disconnect();
  // lists all wallets
  listWallets() -> [Wallet];
}
```

Reply:
@BatiGencho i don't see how the proposed methods can help covering the need `a way to manage the wallet if authorized by the user (add/remove keys)`. I think what @damir is expecting here is a way to import or export key pairs.

@damip can you confirm?
@SlnPons do you have such need on your side ?

Reply:
Ok, so the needs is confirm for two more functions:
- importWallet
- deleteWallet

@BatiGencho I will let you add those two functions with the proper signature inthe markdown file.

Reply:
I simply added importWallet() {...}
  deleteWallet() {...}

just bellow but Provider class is not defined yet.